packages/algob/src/internal/util/lists.ts

```diff
@@ -0,0 +1,27 @@
+
+function getLast<T> (ts: T[]): T {
+  return ts[ts.length - 1];
+}
+
+function getLastSubchild<T> (ts: T[][]): T {
+  return getLast(getLast(ts));
+}
+
+export function partitionByFn<T> (
+  f: (s: T, s1: T) => boolean,
```

Comment:
What does this function do ?

Reply:
It takes a list which has some kind of order
then it also takes a function that checks whether the list should be split there. If it returns true then it should split.
then this:
`f: >` and `[c, d, a, b]`
will return:
`[[c d], [a b]]`

Reply:
This is then later passed to the CP loading method which loads all preceding checkpoints:
For instance I have CPs `1 2 3 4`.
Then I want to execute scripts `2 3 1` which will produce an order of: 
`[[2 3] [1]]`
Then `[2 3]` will:
load CPs `1, 2` then execute script `2`. Then load `3` and execute `3`.
The last `[1]` part will load CP `1` and execute script `1`

Reply:
Okay, understood,

Reply:
what if I have CPs ` 1 2 3 4 5 6 7 ` and 

I want execute ` 6 7 5 4 2 3 1 ` 

will this be splited recursively ?

Reply:
Got it , ` partitionByFn `

Reply:
`6 7 5 2 3 1` ->
[[6 7] [5] [2 3] [1]]